block/manager.go

```diff
@@ -256,42 +256,30 @@ func (m *Manager) ProcessFraudProof(ctx context.Context, cancel context.CancelFu
 	}
 	defer sub.Cancel()
 
-	// continuously process the fraud proofs received via subscription
-	for {
-		// sub.Proof is a blocking call that only returns on proof received or context ended
-		proof, err := sub.Proof(ctx)
-		if err != nil {
-			m.logger.Error("failed to receive gossiped fraud proof", "error", err)
-			return
-		}
+	// blocks until a valid fraud proof is received via subscription
+	// sub.Proof is a blocking call that only returns on proof received or context ended
+	proof, err := sub.Proof(ctx)
+	if err != nil {
+		m.logger.Error("failed to receive gossiped fraud proof", "error", err)
+		return
+	}
 
-		// only handle the state fraud proofs for now
-		fraudProof, ok := proof.(*types.StateFraudProof)
-		if !ok {
-			m.logger.Error("unexpected type received for state fraud proof", "error", err)
-			return
-		}
-		m.logger.Debug("fraud proof received",
-			"block height", fraudProof.BlockHeight,
-			"pre-state app hash", fraudProof.PreStateAppHash,
-			"expected valid app hash", fraudProof.ExpectedValidAppHash,
-			"length of state witness", len(fraudProof.StateWitness),
-		)
-		// TODO(light-client): Set up a new cosmos-sdk app
-		// TODO: Add fraud proof window validation
-
-		success, err := m.executor.VerifyFraudProof(&fraudProof.FraudProof, fraudProof.ExpectedValidAppHash)
-		if err != nil {
-			m.logger.Error("failed to verify fraud proof", "error", err)
-			continue
-		}
-		if success {
-			// halt chain
-			m.logger.Info("verified fraud proof, halting chain")
-			cancel()
-			return
-		}
+	// only handle the state fraud proofs for now
+	fraudProof, ok := proof.(*types.StateFraudProof)
+	if !ok {
+		m.logger.Error("unexpected type received for state fraud proof", "error", err)
+		return
 	}
+	m.logger.Debug("fraud proof received",
+		"block height", fraudProof.BlockHeight,
+		"pre-state app hash", fraudProof.PreStateAppHash,
+		"expected valid app hash", fraudProof.ExpectedValidAppHash,
+		"length of state witness", len(fraudProof.StateWitness),
+	)
+
+	// halt chain
+	m.logger.Info("verified fraud proof, halting chain")
```

Comment:
Recommend Follow Up:

Feels like for a chain halt, we would want to dump as much information as possible, otherwise we need to re-start the chain and re-submit the fraud proof with debug log level to get more useful information. 

I would combine these statements in `info` level so that when/if we have a chain halt we have as much information as possible to react to it.

Reply:
yes, we have a follow-up issue: https://github.com/rollkit/rollkit/issues/933 that will take care of proper halting.